Introduced non-generic type descriptor method `descriptor.Type(System.Type)`.

promontis opened the pull request:
Addresses #1079

michaelstaib commented on src/Core/Types/Types/Descriptors/ArgumentDescriptorBase~1.cs:
```diff
@@ -45,6 +45,20 @@ public void Type<TInputType>()
                 TypeContext.Input);
         }
 
+        public void Type(Type type)
```
The above method should now lean on this non-generic method.

promontis replied:
added

michaelstaib commented on src/Core/Types/Types/Descriptors/OutputFieldDescriptorBase.cs:
```diff
@@ -55,6 +55,20 @@ protected void Type<TOutputType>()
         }
 
+        protected void Type(Type type)
+        {
```
The above method should now lean on this new non-generic method.

promontis replied:
added

promontis commented:
@michaelstaib thanks for having a quick look! I'll add the tests, pull to the interface and resolve you comments! Probably tonight or tomorrow

promontis commented:
@michaelstaib I've added tests, pulled to interface, resolved the comments and fixed a bug.

Is it ok like this?

Btw, I didn't see any `OutputFieldDescriptorTests.cs`. Do you want me to add it?

promontis commented:
@michaelstaib is it ready to be merged :) ? If so, could you also build a preview nuget?

michaelstaib commented:
@promontis sorry was in a meeting ... yes everything is green :) I will trigger the merge and release.
